content/utilities/ms-map-controller/plugin.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * MS Map Controller
+ *
+ * Simple Plugin to pan & zoom the map with keyboard.
+ *
+ * Paning: Arrows (←, ↑, →, ↓)
+ * Zooming: '-' out, '=' in
+ *
+ * Enable logging by setting `window.__MS_MC_DEBUG` to `true` in conosle.
+ *
+ * Author @matthewsimo
+ */
+
+import {
+  html,
+  render,
+} from "https://unpkg.com/htm/preact/standalone.module.js";
+
+class BasePlugin {
+  constructor() {
+    window.__MS_MC_DEBUG = window.__MS_MC_DEBUG || false;
+
+    window.__MS_MC_DEBUG && console.log("[MS MC]: init", { df, ui }, this);
+
+    this.container = null;
+
+    const { heightInWorldUnits, widthInWorldUnits } = ui.getViewport();
+    this.xDelta = widthInWorldUnits / 8; // Distance to move left or right
+    this.yDelta = heightInWorldUnits / 8; // Distance to move up or down
+    this.zoomDelta = 200; // Amount to zoom in or out
+
+    window.addEventListener("keydown", this.handleKeyInput);
+  }
+  async render(container) {
+    this.container = container;
+    render(
+      html`
+        <div>
+          <p>Use arrows to pan.</p>
+          <p>'-' to Zoom Out & '=' to Zoom In.</p>
+        </div>
+      `,
+      container
+    );
```

Comment:
render() could be replaced with

```
    this.div = document.createElement('div');
    this.div.innerHTML = `
        <p>Use arrows to pan.</p>
        <p>'-' to Zoom Out & '=' to Zoom In.</p>
    `;
    container.appendChild(this.div);
```
And the import from the top could be deleted.

This change would reduce the lines of code in the plugin and people no longer have to worry that they have to read the imported files to check for security issues.
Or am I wrong and this and you never have to worry about the imported files?

Reply:
Yep, this is totally valid feedback. My motivation with this plugin is that it can act as a base for someone to take and extend. But if that doesn't align with the goal of the plugins in this repo then I'm happy to strip it down. The functionality is so minimal that it feels like it shouldn't be a standalone plugin TBH though - IMO!